### ISSUE TITLE
modelsbuilder throws exception when property is optional

### ISSUE THREAD

**Issue by ribsdigital** [dawoe/OEmbed-Picker-Property-Editor#11]
hi @dawoe just installed the newest version of the plugin in a new site - lookin good!

one issue i have ran into though is that if a property is not marked as mandatory the models builder blows up...

this is the doctype set up:

![image](https://user-images.githubusercontent.com/22835873/98493639-a8546f80-2286-11eb-8490-f867749092f5.png)

if i select a video, everything works fine - if i leave the property empty, the class generated by the models builder throws the following exception when attempting to view a page that uses the model:

![image](https://user-images.githubusercontent.com/22835873/98493723-e3ef3980-2286-11eb-854f-75882967085c.png)

could this be an issue in the [OEmbedPickerValueConverter](https://github.com/dawoe/OEmbed-Picker-Property-Editor/blob/develop/src/Dawoe.OEmbedPickerPropertyEditor/ValueConverters/OEmbedPickerValueConverter.cs) property converter?

cheers,

jake

**Comment by dawoe** [comment]
Hi @ribsdigital 

I will try to have a look at this today.

Dave

**Comment by dawoe** [comment]
Hi @ribsdigital 

I tried to reproduce this, but I can't get this error.

Can you show me the code in your view ?

This is what I have

```
 @if(Model.SingleVideo != null)
        {
             @Model.SingleVideo.EmbedCode 
        }
```

Dave

**Comment by ribsdigital** [comment]
hi @dawoe thanks for getting back to me 👍 

i did some more digging and i think i've found the issue. if we use the property in a regular doctype, there's no problem, we can do the null check in an if statement and everything works as expected.

however, we're using your package in conjuction with the [https://our.umbraco.com/packages/backoffice-extensions/bento-editor/](https://our.umbraco.com/packages/backoffice-extensions/bento-editor/) package and that's when the exception occurs...

so i downloaded the property editor and dropped it into the project so i could debug - the issue seems to be on this line of the `OEmbedPickerValueConverter` property converter class:

```
if (inter == null)
{
	return allowMultipe ? Enumerable.Empty<OEmbedItem>() : null;
}
```

in a regular doctype, the `inter` property returns as null if a video hasn't been picked. but when it goes through bento, it returns as an empty string...

so i changed the code to this and it now works as expected:

```
if (string.IsNullOrWhiteSpace(inter?.ToString()))
{
	return allowMultipe ? Enumerable.Empty<OEmbedItem>() : null;
}
```

just pushed this up as a pull request ;)

[https://github.com/dawoe/OEmbed-Picker-Property-Editor/pull/12](https://github.com/dawoe/OEmbed-Picker-Property-Editor/pull/12)

**Comment by khraibani** [comment]
Hi @dawoe I am having the issue as well, any chance to get release a version with the fix?
Thank you for the package!

**Comment by dawoe** [comment]
Just released version 4.3.0 that contains this fix
